src/Orleans/Core/GrainFactory.cs

```diff
@@ -290,6 +405,35 @@ internal static TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId
         #endregion
 
         #region Utility functions
+
+        /// <summary>
+        /// Ensures that generated code reflects the current runtime environment.
+        /// </summary>
+        private static void EnsureCodeGenerationIsCurrent()
```

Comment:
Your understanding is correct. I was thinking of changing this method to take a `Type` instead, and only codegen that type's assembly, which is what I do in our `EventJournal` library. I'm not sure I can make that work for the MethodInvokers case, but it will work here.